src/journal/journalctl.c

```diff
@@ -591,30 +595,34 @@ static int parse_argv(int argc, char *argv[]) {
 
                 case ARG_THIS_BOOT:
                         arg_boot = true;
+                        arg_boot_id = SD_ID128_NULL;
+                        arg_boot_offset = 0;
                         break;
 
                 case 'b':
-                        arg_boot = true;
+                        arg_boot_id = SD_ID128_NULL;
+                        arg_boot_offset = 0;
```

Comment:
please move these lines into the final, `arg_boot = true` branch of the if statement. After all, if there's success in the first two branches, then these fields are reset anyway. And if they fail, then there's no point in overriding these fails. Thus only in the third branch case it makes sense to reset this.

Reply:
I was going to make this change too, but when i went back to it i realised that the `else` thing is flawed — it leaves unaccounted-for the condition where `-b` is used, and there is a following argument, but it isn't a boot descriptor, as in `journalctl -ball -b -q`. In that case, `-b` will do nothing at all again (rather than act like `-b0`). So i moved all of the re-initialisation stuff back up to the top (and i added a test to catch that case i'd initially missed). Please let me know if you think there's a better way after all